src/app/settings/keyboard-mapping-settings/components/edit-keyboard-mapping-dialog/edit-keyboard-mapping-dialog.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, Inject } from '@angular/core'
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog'
+import { KeyboardMapping } from '../keyboard-mapping-settings-page/keyboard-mapping-settings-page.component'
+
+@Component({
+  // There is no selector because this component is only allowed to be called through the DialogService.
+  templateUrl: './edit-keyboard-mapping-dialog.component.html',
+})
+export class EditKeyboardMappingDialogComponent {
+  public readonly keyboardMapping?: KeyboardMapping
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) keyboardMapping: KeyboardMapping,
+    private readonly dialogRef: MatDialogRef<EditKeyboardMappingDialogComponent>
```

Comment:
Since we still utilize the Material Sidebar for creating at the actual sidebar, we still need some way to get the "input" variable of the component. This is done by the token injection. We also need to be able to close the dialog, this is done by the DialogRef. 
Then DialogService has no idea of what components it is rendering in as a Sidebar and therefore has no way of knowing when to close the sidebar (not to mention it also has no way of receiving the result of the sidebar).

Reply:
I don't think you should spend much time on it. We have more important things to do ;)